ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -918,8 +918,21 @@ class JITModBuilder {
     CallInst *CPCI = Builder.CreateCall(Func, {});
     Builder.CreateUnreachable();
 
-    JITModBuilder JB(AOTMod, FuncNames, BBs, TraceLen, FAddrKeys, FAddrVals,
-                     FAddrLen, CPCI, TraceInputs);
+    // Populate the function address map with dummy entries for all of the
+    // functions in the AOT module, so that the trace compiler can outline
+    // calls to them if neccessary.
+    //
+    // The actual addresses inserted don't matter, as the trace compiler suite
+    // only compiles traces (without executing them).
+    std::vector<char *> NewFAddrKeys;
+    std::vector<void *> NewFAddrVals;
+    for (Function &F : AOTMod->functions()) {
+      NewFAddrKeys.push_back(const_cast<char *>(F.getName().data()));
+      NewFAddrVals.push_back((void *)0xdeadbeef);
```

Comment:
I would tend to prefer `0` here (since if anyone dereferences that, it's more likely to segfault than 0xdeadbeef which is a valid virtual address location).

Reply:
Sadly we can't do this. The entries we are adding to this map eventually get passed to `ExecutionEngine::addGlobalMapping()` and (although this isn't documented)  if you pass `NULL` pointers to it, then this has special meaning to LLVM: that the symbol address is unknown.

If we try to point symbols to NULL, LLVM will halt compilation and return an error message like:
```
Symbol not found: f
```

So we have to use a non-null value. Would `0x1` be preferable perhaps?

Reply:
`0x1` sounds like a reasonable compromise.